contracts/proposal/dao-proposal-multiple/src/contract.rs

```diff
@@ -263,11 +251,42 @@ pub fn execute_propose(
 
     let hooks = new_proposal_hooks(PROPOSAL_HOOKS, deps.storage, id, proposer.as_str())?;
 
+    let sender = info.sender.clone();
+
+    // Auto cast vote if given.
+    let (vote_hooks, vote_attributes) = if let Some(vote) = vote {
+        let response = execute_vote(
+            deps,
+            env,
+            proposer.clone(),
+            id,
+            vote.vote,
+            vote.rationale.clone(),
+        )?;
+        (
+            response.messages,
+            vec![
+                Attribute {
+                    key: "position".to_string(),
+                    value: vote.vote.to_string(),
+                },
+                Attribute {
+                    key: "rationale".to_string(),
+                    value: vote.rationale.unwrap_or_else(|| "_none".to_string()),
+                },
+            ],
+        )
+    } else {
+        (vec![], vec![])
+    };
+
     Ok(Response::default()
         .add_submessages(hooks)
+        .add_submessages(vote_hooks)
```

Comment:
curious about one thing here: given that we basically combine the propose & vote actions into one, why not submit the vote hooks as a regular (fire and forget) message? in that case if proposal creates successfully, but the vote fails (e.g. voting on a non-existent option), the proposal would still be created fine. somewhat mimicking the current way to propose and vote.

_fwiw i also think this should be a submessage though_

Reply:
hmm, i don't think it would be good practice to partially succeed a message. in my mind, if the user intends to submit a proposal and vote on it, it should only succeed if what they intended to happen actually happens. there might be a specific reason/workflow they've built that requires proposing and voting to occur atomically. if you allow votes to fail, you're essentially allowing invalid inputs to be submitted.

also, voting fails anyways in the `execute_vote` function call above this part. these submessages are just the hooks that are triggered after voting succeeds.